### ISSUE TITLE
Python3.6 Packages

### ISSUE THREAD

**Issue by Miserlou** [Miserlou/lambda-packages#43]
This is annoying. Is everything going to have to be repackaged for 3.6?

We'll have to change the manifest format as welll, suggest we simply add another layer so that the runtime is before the package name.

**Comment by nikbora** [comment]
I am happy to do the "pre-pend runtime" update for current python 2.7 packages as am updating that bit of code in Zappa anyways. For 3.6 I suggest to only recompile those that don't have a wheels version for the particular package version, perhaps via a separate PR per new package.

Just to be clear before proceeding, would the following work?

## Package file name

Same directory name but instead of filename being something like `pycrypto-2.6.1.tar.gz` we now have `python2.7-pycrypto-2.6.1.tar.gz`

## Manifest

Instead of:

```
'psycopg2': {
        'version': '2.6.1',
        'path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                            'psycopg2', 'psycopg2-2.6.1.tar.gz')
    },
```
we have:

```
'psycopg2': {
        'version': '2.6.1',
        'path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                            'psycopg2', 'python2.7-psycopg2-2.6.1.tar.gz')
    },
'python2.7-psycopg2': {
        'version': '2.6.1',
        'path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                            'psycopg2', 'python2.7-psycopg2-2.6.1.tar.gz')
    },
```

I am suggesting to keep the old entry which defaults to python2.7 for backward compatibility with anyone who is using this outside of Zappa, but mention in docs to use the new format.

@Miserlou whacha say?

**Comment by nikbora** [comment]
Or perhaps even simpler:

```
'psycopg2': {
        'version': '2.6.1',
        'path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                            'psycopg2', 'python2.7-psycopg2-2.6.1.tar.gz'),
        'python2.7-path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                            'psycopg2', 'python2.7-psycopg2-2.6.1.tar.gz')
    },
```

**Comment by Miserlou** [comment]
Don't like this method at all, sorry. It's redundant and non-semantic. Also, backwards compatibility is not a design requirement here. Additionally, we may have situations where the P2 and P3 versions of a package differ. 

So, we have two choices:

```python
{
'python2.7': {
    'psycopg2': {
        'version': '2.6.1',
        'path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                                'psycopg2', 'python2.7-psycopg2-2.6.1.tar.gz'),
    }
},
'python3.6': {
    'psycopg2': {
        'version': '3.0.0',
        'path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                                'psycopg2', 'python3.6-psycopg2-3.0.0.tar.gz'),
    }
},
```

OR

```python
{
'psycopg2': {
    'python2.7': {
        'version': '2.6.1',
        'path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                                'psycopg2', 'python2.7-psycopg2-2.6.1.tar.gz'),
        }
    },
    'python3.6': {
        'version': '3.0.0',
        'path': os.path.join(os.path.dirname(os.path.abspath(__file__)),
                                'psycopg2', 'python3.6-psycopg2-3.0.0.tar.gz'),
        }
    }
}
```

Looking at it now, I think I prefer the second.

**Comment by nikbora** [comment]
Agree. Have thought of the same but was under impression backwards compatibility was important. If not then this is much nicer, will update PR.

**Comment by nikbora** [comment]
Done: https://github.com/Miserlou/lambda-packages/pull/45

**Comment by readevalprint** [comment]
Thanks for the work you all have done on this because this problem just hit me. I'd be happy to help test it.

**Comment by unux-git** [comment]
#45 merged, why issue still open, is that not full support yet?

**Comment by Miserlou** [comment]
Yeah, this can be changed, it landed in Zappa 0.41.2.
